sdks/python/setup.py

```diff
@@ -146,7 +146,7 @@ def get_version():
     'grpcio>=1.29.0,<2',
     'hdfs>=2.1.0,<3.0.0',
     'httplib2>=0.8,<0.18.0',
-    'mock>=1.0.1,<3.0.0',
```

Comment:
I think this should really be a test-only dependency, but it predates us separating those out. Would it also unblock you if we move this to `REQUIRED_TEST_PACKAGES` so it's not required by default?

Reply:
Yes I think move this to REQUIRED_TEST_PACKAGES is a good idea. Thank you!

Reply:
That would also help us enormously in Apache Airflow . Currently `apache-beam` package is the only package excluded from installation on our CI because it has the mock and dill dependency pinned to very old versions:

https://github.com/apache/airflow/blob/master/setup.py#L743

```
# Those packages are excluded because they break tests and they are not needed to run our test suite.
# This can be removed as soon as we get non-conflicting
# requirements for the apache-beam as well.
#
# Currently Apache Beam has very narrow and old dependencies for 'dill' and 'mock' packages which
# are required by our tests (but only for tests).
#
PACKAGES_EXCLUDED_FOR_CI = [
    'apache-beam',
]
```

I think dill is not a problem (we only use it with beam tests and 0.3.3 is anyhow the latest available version), but moving mock to test packages would help enormously.

Reply:
Thanks @potiuk, good to know this will help airflow as well :)

I filed BEAM-12069 to track the issue fixed by this PR, BEAM-11167 would be the appropriate jira for the dill issue. I don't think there's any reason not to go ahead and upgrade it, we just have it pinned to a narrow range to avoid compatibility issues (e.g. see BEAM-4785). Would you be interested in trying to upgrade dill to 0.3.2 or 0.3.3?

Reply:
I think it is fine to upgrade dill. Although we will still need a very narrow range for dill for this reason: https://github.com/apache/beam/blob/master/sdks/python/setup.py#L138